gather_istio.sh

```diff
@@ -0,0 +1,224 @@
+#!/bin/bash
+set -ex
+# Copyright 2024 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+BASE_COLLECTION_PATH="/tmp/must-gather"
+
+# Get the namespaces of all control planes in the cluster
+function getControlPlanes() {
+  local result=()
+
+  local namespaces
+  #TODO: there should be never Istiorevision in a namespaces without Istio resource??
```

Comment:
Users can also create IstioRevisions directly (i.e. without creating Istio), but this is a very special case.